File: Longest Substring Without Repeating Characters.py
class Solution:
    # @return an integer
    def lengthOfLongestSubstring(self, s):
        myMap={}
        start=0
        max=0
        for i in range(len(s)):
            if s[i] in myMap:
                for num in range(start,myMap[s[i]]):
                    myMap.pop(s[num])
                start=myMap[s[i]]+1
                myMap[s[i]]=i
            else:
                myMap[s[i]]=i
                if i-start+1>max:
                    max=i-start+1
        return max

File: Cpp/1-10/Longest Substring Without Repeating Characters.cpp

/**
 * 16ms
 * error: 1  
 * 
 * 
 */ 
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int beg=0,Max=0;
        vector<int> Map(128,-1);
        for(int i=0; i<s.size(); i++){
          if(Map[s[i]]>=beg){
            Max = max(Max,i-beg);
            beg=Map[s[i]]+1;
          }
          Map[s[i]]=i;                //error 1:  add else first time
        }
        Max =max(Max,(int)s.size()-beg);
        return Max;
    }
};
